server/store.ml

```diff
@@ -53,7 +53,7 @@ module Node = struct
 
   let replace_child node child nchild =
     (* this is the on-steroid version of the filter one-replace one *)
-    let rec replace_one_in_list l =
+    let[@tail_mod_cons] rec replace_one_in_list l =
```

Comment:
This raises the requirement for OCaml from 4.08 to 4.14. This will need a change in `xenstore.opam`. Is the current minimum version needed for some of the users? (for xapi 4.14 is used currently)